kernel/main.c

```diff
@@ -340,6 +353,52 @@ async_init(void)
       panic("Unable to create a kthread for do_async_init()");
 }
 
+static void
+setup_uefi_runtime_services() {
+   EFI_RUNTIME_SERVICES *RT = TO_PTR(uefi_rt_addr);
+
+   printk("Runtime Services: %p\n", RT);
+
+   struct mem_region ma;
+
+   EFI_MEMORY_DESCRIPTOR *virt_map = kmalloc((size_t)get_mem_regions_count());
+   int num_entries_virt_map = 0;
+
+   for (int i = 0; i < get_mem_regions_count(); i++) {
+      get_mem_region(i, &ma);
+
+      if (ma.type == TILCK_BOOT_EFI_RUNTIME_CODE_REQ || ma.type == TILCK_BOOT_EFI_RUNTIME_DATA_REQ) {
```

Comment:
Yeah, if only those memory types are really needed, we might just drop the other ones.

Reply:
Also, cosmetic. Avoid too many indentations by keeping the main code path outside an `if`. Just:
```
if (ma.type != CODE && ma.type != DATA)
   continue;
```